<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Mocha Tests" time="0.211" tests="16" failures="11">
  <testsuite name="Root Suite" timestamp="2024-06-25T17:48:15" tests="0" time="0.000" failures="0">
  </testsuite>
  <testsuite name="stock_trades_api_hard" timestamp="2024-06-25T17:48:15" tests="16" file="F:\Test\Eiger\SUbmit\Eiger-Test\2. NodeJS Stock Trades API\project\test\index.spec.js" time="0.209" failures="11">
    <testcase name="stock_trades_api_hard should create a new trade of buy type" time="0.028" classname="should create a new trade of buy type">
    </testcase>
    <testcase name="stock_trades_api_hard should create a new trade of sell type" time="0.006" classname="should create a new trade of sell type">
    </testcase>
    <testcase name="stock_trades_api_hard should return 400 if the payload has an invalid type" time="0.003" classname="should return 400 if the payload has an invalid type">
    </testcase>
    <testcase name="stock_trades_api_hard should return 400 if the number of shares exceeds the limit" time="0.003" classname="should return 400 if the number of shares exceeds the limit">
    </testcase>
    <testcase name="stock_trades_api_hard should return 400 if the number of shares is below the limit" time="0.003" classname="should return 400 if the number of shares is below the limit">
    </testcase>
    <testcase name="stock_trades_api_hard should fetch all the trades" time="0.023" classname="should fetch all the trades">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at Context.<anonymous> (test\index.spec.js:106:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch no trades if the type filter value does not exist" time="0.018" classname="should fetch no trades if the type filter value does not exist">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at Context.<anonymous> (test\index.spec.js:113:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch all trades for a user" time="0.016" classname="should fetch all trades for a user">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at Context.<anonymous> (test\index.spec.js:120:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch no trades if user filter value does not exist" time="0.015" classname="should fetch no trades if user filter value does not exist">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at Context.<anonymous> (test\index.spec.js:127:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch all buy trades for a user" time="0.014" classname="should fetch all buy trades for a user">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at Context.<anonymous> (test\index.spec.js:134:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch all sell trades for a user" time="0.013" classname="should fetch all sell trades for a user">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at Context.<anonymous> (test\index.spec.js:141:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch a single trade" time="0.005" classname="should fetch a single trade">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at Context.<anonymous> (test\index.spec.js:148:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should get 404 if the trade ID does not exist" time="0.002" classname="should get 404 if the trade ID does not exist">
      <failure message="expected &apos;&lt;!DOCTYPE html&gt;\n&lt;html lang=&quot;en&quot;&gt;\n&lt;h…&apos; to deeply equal &apos;ID not found&apos;" type="AssertionError"><![CDATA[AssertionError: expected '<!DOCTYPE html>\n<html lang="en">\n<h…' to deeply equal 'ID not found'
    at Context.<anonymous> (test\index.spec.js:155:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -<!DOCTYPE html>
      -<html lang="en">
      -<head>
      -<meta charset="utf-8">
      -<title>Error</title>
      -</head>
      -<body>
      -<pre>Cannot GET /trades/32323</pre>
      -</body>
      -</html>
      +ID not found
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should get 405 for a put request to /trades/:id" time="0.006" classname="should get 405 for a put request to /trades/:id">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 405 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 405 but got 404
    at Context.<anonymous> (test\index.spec.js:161:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -404
      +405
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should get 405 for a patch request to /trades/:id" time="0.005" classname="should get 405 for a patch request to /trades/:id">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 405 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 405 but got 404
    at Context.<anonymous> (test\index.spec.js:167:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -404
      +405
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should get 405 for a delete request to /trades/:id" time="0.005" classname="should get 405 for a delete request to /trades/:id">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 405 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 405 but got 404
    at Context.<anonymous> (test\index.spec.js:173:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

      + expected - actual

      -404
      +405
      ]]></failure>
    </testcase>
  </testsuite>
</testsuites>